src/daria/analysis/segmentationcomparison.py

```diff
@@ -1,113 +1,471 @@
 from __future__ import annotations
 
+from typing import Optional
+
+import cv2
+import matplotlib.patches as mpatches
+import matplotlib.pyplot as plt
 import numpy as np
+import skimage
 from matplotlib.cm import get_cmap
 
+import daria as da
+
 
-def compare_segmentations(*segmentations, **kwargs) -> np.ndarray:
+class CompareSegmentations:
     """
-    Comparison of segmentations.
-
-    Args:
-        segmentations (asterisk argument): Allows to provide an arbitraty
-                number of segmented numpy arrays of integers to be compared
-        optional keyword arguments (kwargs):
-            components (list): list of the different components that are
-                considered in the segmented images. So far only two are allowed
-                to be provided.
-            non_active_component (int): value of nonactive component in the segmented
-                images.
-            gray_colors (np.ndarray): array of three different scales of
-                gray (in RGB format), one for each of the different combinations of
-                components in the segmentations.
-            colors (np.ndarray): Array of different colors that should
-                indicate uniqe components in each segmentation.
-            light_scaling (float): Indicate how much lighter the second
-                component should be scaled in its unique color.
+    Class for comparing segmented images.
+
+    Routines for comparing segmentations and creating visualizations of the comparison.
+
+    Attributes:
+        number_of_segmented_images (int): Number of segmented images that one compares
+        segmentation_names (list[str]): list of names for each of the segmentations.
+            Will affect legends in plots.
+        components (list): list of values the different (active) components in the
+            segmentations. As of now, up to two are allowed for and the default values
+            are 1 and 2.
+        non_active_component (int): value for the nonactive component in the
```

Comment:
why is this needed, when `components` needs to be provided? Are not all other components inactive?

Reply:
Sounds very correct, ill remove it!